app/src/backend/environment/utils.ts

```diff
@@ -87,3 +87,9 @@ export const mergeLocalAndSyncVariables = (
     ...updatedVariables,
   };
 };
+
+export const extractVariableNameFromStringIfExists = (string: string) => {
+  const regex = /{{([^}]+)}}/;
+  const match = string.match(regex);
```

Comment:
Please cross check it once, if it matches all the variables in the string or only the first occurrence.
eg: `{{test}}-some-string-{{test}}`, will this be a case? if not then can be ignored